app/models/changeset/pull_request.rb

```diff
@@ -3,11 +3,14 @@ class Changeset::PullRequest
   RISKS_SECTION = /#+\s+Risks.*\n/i
 
   # Matches URLs to JIRA issues.
-  JIRA_ISSUE = %r[https://\w+\.atlassian\.net/browse/[\w-]+]
+  JIRA_ISSUE_URL = %r[https?:\/\/[\da-z\.\-]+\.[a-z\.]{2,6}\/browse\/[A-Z][A-Z\d]+-\d+]
```

Comment:
There are two intentions in the JIRA issue linked to this PR. My bad I should have mentioned both.

The other intention behind the change is to support both JIRA Cloud (which would be hosted on *.atlassian.net) and JIRA Server (which could be hosted on any hostname) by default. So I think that the generic (non-JIRA_URL-injected) regex should still be there by default.

I guess the case could be made for conditionally injecting the regex, but that would be a design decision.

The current implementation is in line with the relevant description in the linked JIRA issue which I would quote here:
 
> If a full JIRA url is mentioned, we would take that full url. Otherwise, the issue key found will be prefixed with the base url configured below